src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -464,9 +476,12 @@ private void parseMasterEventMetrics(String rootLocation, long currWindowStartTi
 
         long mFinalT = System.currentTimeMillis();
         LOG.info("Total time taken for parsing Master Event Metrics: {}", mFinalT - mCurrT);
+        timingStats.put("parseMasterEventMetrics", (double)(mFinalT - mCurrT));
     }
 
     public void processMetrics(String rootLocation, long currTimestamp) throws Exception {
+        timingStats = new HashMap<>();
```

Comment:
How about clearing the contents of timeStats instead of creating a new one every 2.5 seconds?  This can save some heap memory allocation.

Reply:
This way it is easy to manage concurrency